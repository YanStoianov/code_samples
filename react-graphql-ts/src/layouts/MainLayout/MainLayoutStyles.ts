import styled from "styled-components";

export default styled.div`
  .main-layout {
    .layout-content {
      min-height: calc(100vh - 64px - 70px);
    }

    .layout-logo {
      max-height: 54px;
    }

    .layout-header {
      display: flex;
      align-items: center;

      .layout-title {
        margin: 0 auto 0 auto;
        color: #fcfcfc;
        font-size: 30px;
      }
    }

    .layout-footer {
      display: flex;
      justify-content: center;
    }

    .approved {
      color: green;
    }

    .disapproved {
      color: red;
    }
  }
`;