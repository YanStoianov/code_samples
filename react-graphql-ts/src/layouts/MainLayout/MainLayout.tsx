import React from 'react';
import {Layout} from 'antd';
import {MainLayoutProps} from "./types";
import MainLayoutStyles from "./MainLayoutStyles";

const {Header, Content, Footer} = Layout;
const logo = require("../../assets/pokedex-icon.jpg")


const MainLayout = (props: MainLayoutProps) => {
    const {children} = props;

    return (
        <MainLayoutStyles>
            <Layout className={'main-layout'}>
                <Header className={'layout-header'}>
                    <img src={logo} alt="pokedex" className='layout-logo'/>
                    <h1 className={'layout-title'}>Pokedex</h1>
                </Header>
                <Content className={'layout-content'}>
                    {children}
                </Content>
                <Footer className={'layout-footer'}>Â©2022 Created by Vlad</Footer>
            </Layout>
        </MainLayoutStyles>
    )
};

export default MainLayout;