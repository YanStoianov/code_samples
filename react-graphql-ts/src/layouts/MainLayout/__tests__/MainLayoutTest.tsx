import React from 'react'
import {render} from '@testing-library/react';
import 'jest-styled-components'
import MainLayout from "../MainLayout";

jest.mock('../../../../assets/pokedex-icon.jpg')

describe("MainLayoutTest", () => {
    beforeAll(() => {
        Object.defineProperty(window, "matchMedia", {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(),
                removeListener: jest.fn(),
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            }))
        });
    });

    test('snapshot', () => {
        const {asFragment} = render(
            <MainLayout>
                <div>test</div>
            </MainLayout>
        );
        expect(asFragment()).toMatchSnapshot();
    })
});