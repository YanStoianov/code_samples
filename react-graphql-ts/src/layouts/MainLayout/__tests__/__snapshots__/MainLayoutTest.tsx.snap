// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayoutTest snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe hFNikj"
  >
    <section
      class="ant-layout main-layout"
    >
      <header
        class="ant-layout-header layout-header"
      >
        <img
          alt="pokedex"
          class="layout-logo"
          src="[object Object]"
        />
        <h1
          class="layout-title"
        >
          Pokedex
        </h1>
      </header>
      <main
        class="ant-layout-content layout-content"
      >
        <div>
          test
        </div>
      </main>
      <footer
        class="ant-layout-footer layout-footer"
      >
        Â©2022 Created by Vlad
      </footer>
    </section>
  </div>
</DocumentFragment>
`;
